controller/node_controller.go

```diff
@@ -995,6 +1003,178 @@ func (nc *NodeController) syncNodeStatus(pod *corev1.Pod, node *longhorn.Node) e
 	return nil
 }
 
+func (nc *NodeController) environmentCheck(kubeNode *corev1.Node, node *longhorn.Node) error {
+	if err := nc.syncKernelVersion(kubeNode, node); err != nil {
+		return err
+	}
+
+	namespaces := []lhtypes.Namespace{lhtypes.NamespaceMnt, lhtypes.NamespaceNet}
+	if err := nc.syncPackagesInstalled(kubeNode, node, namespaces); err != nil {
+		return err
+	}
+
+	if err := nc.syncMultipathd(node, namespaces); err != nil {
+		return err
+	}
+
+	if err := nc.syncNFSClientVersion(kubeNode, node, namespaces); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (nc *NodeController) syncKernelVersion(kubeNode *corev1.Node, node *longhorn.Node) error {
+	kernelVersion := kubeNode.Status.NodeInfo.KernelVersion
+	brokenKernelVersion := []string{"5.15.0", "6.5.6"}
+	majorKernel5150Version := 5
+	majorKernel656Version := 6
+	brokenKernel5150PatchVersion := 94
+	brokenKernel656PatchVersion := 6
+	fixedKernel5150PatchVersion := 100
+
+	brokenFlag := false
+	for _, version := range brokenKernelVersion {
+		if !strings.Contains(kernelVersion, version) {
+			continue
+		}
+		var kernel, major, minor, patch int
+		if _, err := fmt.Sscanf(kernelVersion, "%d.%d.%d-%d", &kernel, &major, &minor, &patch); err != nil {
+			return errors.Wrap(err, "failed to parse kernel version "+kernelVersion)
+		}
```

Comment:
log the error in the condition. WDYT?

Reply:
Logged the error to NodeConditionTypeSupportedKernel condition. Thanks